packages/imt-rs/src/lib.rs

```diff
@@ -0,0 +1,320 @@
+use std::error::Error;
+
+use tiny_keccak::{Hasher, Keccak};
+
+pub struct IMT {
+    nodes: Vec<Vec<Option<IMTNode>>>,
+    zeroes: Vec<IMTNode>,
+    hash: IMTHashFunction,
+    depth: usize,
+    arity: usize,
+}
+
+struct IMTMerkleProof {
```

Comment:
should we mark this as `public` too? seems treated as private type here but thought as public. not really sure, forwarding to @curryrasul